numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -932,6 +938,8 @@ typedef int (PyArray_FinalizeFunc)(PyArrayObject *, PyObject *);
 #define NPY_ARRAY_UPDATEIFCOPY    0x1000 /* Deprecated in 1.14 */
 #define NPY_ARRAY_WRITEBACKIFCOPY 0x2000
 
+#define NPY_ARRAY_ENSURENOCOPY 0x4000
```

Comment:
I know it is bound to get outdated, but maybe we should add a comment which functions honor this?  I guess most do, but it seems things are a bit confusing with many of these flags just honored by the `CheckFromAny` function and not by the `FromAny` version...